File: IronPythonSilverlight/highscore.py

from System.IO import FileMode, StreamWriter, StreamReader
from System.IO.IsolatedStorage import IsolatedStorageFile, IsolatedStorageFileStream

class HighScore:
    def __init__(self):
        self.Level = 0
        self.Name = 'Nobody'
        self.LoadHighScore()

    def getMessage(self):
        return 'Level {0} by {1}'.format(self.Level, self.Name)

    Message = property(getMessage)

    def Save(self):
        filename = "record.txt"
        data = "{0}:{1}".format(self.Level,self.Name)
        with IsolatedStorageFile.GetUserStoreForApplication() as isf:
            with IsolatedStorageFileStream(filename, FileMode.Create, isf) as isfs:
                with StreamWriter(isfs) as sw:
                    sw.Write(data)

    def LoadHighScore(self):
        filename = "record.txt"
        with IsolatedStorageFile.GetUserStoreForApplication() as isf:
            if not isf.FileExists(filename):
                return
            with IsolatedStorageFileStream(filename, FileMode.Open, isf) as isfs:
                with StreamReader(isfs) as sr:
                    record = sr.ReadLine()
                    parts = record.split(':')
                    if(len(parts) == 2):
                        self.Level = int(parts[0])
                        self.Name = parts[1]        

File: Silverlight4/Asterisk/HighScore.cs
using System;
using System.IO;
using System.IO.IsolatedStorage;

namespace Asterisk
{
    public class HighScore
    {
        public int Level { get; set; }
        public string Name { get; set; }
        private static readonly string filename = "record.txt";

        public HighScore()
        {
            Level = 0;
            Name = "Nobody";
            LoadHighScore();
        }

        public string Message
        {
            get
            {
                return String.Format("Level {0} by {1}", this.Level, this.Name);
            }
        }

        public void Save()
        {
            
            string data = String.Format("{0}:{1}",this.Level,this.Name);
            using (var isf = IsolatedStorageFile.GetUserStoreForApplication())
            {
                using (var isfs = new IsolatedStorageFileStream(filename, FileMode.Create, isf))
                {
                    using (var sw = new StreamWriter(isfs))
                    {
                        sw.Write(data);
                    }
                }
            }
        }

        private void LoadHighScore()
        {
            using (var isf = IsolatedStorageFile.GetUserStoreForApplication())
            {
                if (!isf.FileExists(filename))
                {
                    return;
                }

                using (var isfs = new IsolatedStorageFileStream(filename, FileMode.Open, isf))
                {
                    using (var sr = new StreamReader(isfs))
                    {
                        var record = sr.ReadLine();
                        var parts = record.Split(':');
                        if (parts.Length == 2)
                        {
                            this.Level = int.Parse(parts[0]);
                            this.Name = parts[1];
                        }
                    }
                }
            }
        }
    }
}
